src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -1926,6 +1926,32 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                 break;
             }
 
+            case NI_Sve_SaturatingDecrementByActiveElementCount:
+            case NI_Sve_SaturatingIncrementByActiveElementCount:
+            {
+                // RMW semantics
+                if (targetReg != op1Reg)
+                {
+                    assert(targetReg != op2Reg);
+                    GetEmitter()->emitIns_Mov(INS_mov, emitTypeSize(node), targetReg, op1Reg, /* canSkip */ true);
+                }
+
+                // Switch instruction if arg1 is unsigned.
+                if (varTypeIsUnsigned(node->GetAuxiliaryType()))
```

Comment:
shouldn't this also use the unsigned variant of the instruction i.e. the first arg is `Vector<ulong>` and hence, the base type is `ulong`.

```c#
public static System.Numerics.Vector<ulong> SaturatingDecrementByActiveElementCount(System.Numerics.Vector<ulong> value, System.Numerics.Vector<ulong> from) { throw null; }
```

Reply:
Aux type is set as:
`retNode->AsHWIntrinsic()->SetAuxiliaryJitType(op1BaseJitType);`

So, for the example, `GetAuxiliaryType()` should be returning `ulong`

Reply:
ah, right.